### PR TITLE
Fix for lost clipboard contents (#5424)

### PR DESCRIPTION
**Pull request by gavynriebau**
Fixes the issue described in #5424 

The code has been changed so that for unix type platforms a call is made to `libc::setsid` using the unix specific CommandExt [pre_exec](https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.pre_exec).

The `setsid` call happens just after `fork` has been called and before `execve` is called. This means it runs in the context of the child process.

The call is necessary to ensure that spawned child processes related to the clipboard (e.g. `xclip`) have a distinct process session ID and process group ID.

Without this change when the parent process exits, the child (i.e. `xclip`), can be erroneously killed leading to copy/paste breaking in some specific circumstances such as those detailed in #5424

### CODE REVIEW

File: helix-view/src/clipboard.rs [hunk 1]
```diff
@@ -276,12 +276,27 @@ pub mod provider {
...
+                if cfg!(not(any(windows, target_os = "wasm32", target_os = "macos"))) {
+                    use std::os::unix::process::CommandExt;
+
+                    unsafe {
+                        command_mut = command_mut.pre_exec(|| match libc::setsid() {
+                            -1 => Err(std::io::Error::last_os_error()),
+                            _ => Ok(()),
+                        });
+                    }
+                }
```
Comment (archseer) [inline helix-view/src/clipboard.rs:297]: `setsid` should be called on all unix systems:

```suggestion
                #[cfg(unix)]
                unsafe {
                    use std::os::unix::process::CommandExt;
                    command_mut = command_mut.pre_exec(|| {
                        if libc::setsid() == -1 {
                            return Err(io::Error::last_os_error());
                        }
                    });
                }
```

For windows we might also want to pass `.creation_flags(CREATE_NEW_PROCESS_GROUP)`

**Review by archseer — CHANGES_REQUESTED** [review]

File: helix-view/src/clipboard.rs (hunk 1 above)
Comment (gavynriebau) [reply helix-view/src/clipboard.rs]: I've made the following changes:
- Call `setsid` for all unix systems
- Call `.creation_flags(CREATE_NEW_PROCESS_GROUP)` for windows

After making the changes I've realised the change to set `CREATE_NEW_PROCESS_GROUP` will never be executed as it is inside a `mod provider` that uses the config `#[cfg(not(target_os = "windows"))]`.

Seeking your opinion but I am thinking of reverting back by one commit so that the PR will contain no code related to setting `CREATE_NEW_PROCESS_GROUP` for windows but keeps the change you suggested of calling `setsid` for all unix systems?

Separate to this PR but I'm also considering doing a follow-up PR to cleanup the code a little, let me know if you think this would be valuable?
With the amount of conditional compilation attributes I found it slightly hard to follow what was going on andwas thinking of splitting the code out into separate files for each environment, e.g. a provider file for windows, macos, osc52, wasm, unix etc..

File: helix-view/Cargo.toml [hunk 2]
```diff
@@ -22,6 +22,7 @@ helix-lsp = { version = "0.6", path = "../helix-lsp" }
 helix-dap = { version = "0.6", path = "../helix-dap" }
 crossterm = { version = "0.25", optional = true }
 helix-vcs = { version = "0.6", path = "../helix-vcs" }
+libc = "0.2"
```
Comment (archseer) [inline helix-view/Cargo.toml:25]: Should this be listed under `[target.'cfg(unix)'.dependencies`? It probably doesn't make a difference

Comment (gavynriebau) [reply helix-view/Cargo.toml]: Thanks, yes, you're right, it's a unix only dependency. I've addressed this now.

File: helix-view/src/clipboard.rs (hunk 1 above)
Comment (gavynriebau) [reply helix-view/src/clipboard.rs]: I think these changes are ready for another review now.

I'm not able to change how `pre_exec` is called to exactly as you suggested because it needs to return a `Result<()>` type.

**Review by archseer — APPROVED** [review]
